plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/parameterInfo/KotlinHighLevelFunctionParameterInfoHandler.kt

```diff
@@ -447,7 +468,8 @@ abstract class KotlinHighLevelParameterInfoWithCallHandlerBase<TArgumentList : K
         val allParametersUsed = usedParameterIndices.size == valueParameterCount
         val supportsTrailingCommas = callElement.languageVersionSettings.supportsFeature(LanguageFeature.TrailingCommas)
         val afterTrailingComma = arguments.isNotEmpty() && currentArgumentIndex == arguments.size
-        val tooManyArgs = allParametersUsed && ((!supportsTrailingCommas && afterTrailingComma) || arguments.size > valueParameterCount)
+        val isInPositionToEnterArgument = !supportsTrailingCommas && afterTrailingComma
+        val tooManyArgs = allParametersUsed && (isInPositionToEnterArgument || hasUnmappedArgument)
```

Comment:
Just to double-check, replacement with `hasUnmappedArgument` is equivalent to the previous logic---`arguments.size > valueParameterCount` only if it's `&&`-ed with `allParametersUsed` I guess? Well, if `tooManyArgs` is only used for `isDisabled`, having any unmapped argument is a good qualifier(?), but it seems not sufficient to declare there are too many args (unless all parameters are used).

Reply:
I think I understand the first part of your comment... Yes, I'm replacing `arguments.size > valueParameterCount` with `hasUnmappedArgument`. The former doesn't work when there are multiple arguments for a `vararg` parameter.

But I'm not sure I understand the second part of your comment "Well if `tooManyArgs`..." Can you clarify? There are multiple conditions for `isDisabled`, having too many args is just one of them. You can only have "too many args" if all parameters are used.